lib/src/provider.dart

```diff
@@ -287,42 +286,22 @@ The context used was: $context
     return inheritedElement.value;
   }
 
-  static _InheritedProviderScopeElement<T> _inheritedElementOf<T>(
+  static _InheritedProviderScopeElement<T>? _inheritedElementOf<T>(
     BuildContext context,
   ) {
-    assert(context != null, '''
-Tried to call context.read/watch/select or similar on a `context` that is null.
-
-This can happen if you used the context of a StatefulWidget and that
-StatefulWidget was disposed.
-''');
-    assert(
-      _debugIsSelecting == false,
-      'Cannot call context.read/watch/select inside the callback of a context.select',
-    );
-    assert(
-      T != dynamic,
-      '''
-Tried to call Provider.of<dynamic>. This is likely a mistake and is therefore
-unsupported.
-
-If you want to expose a variable that can be anything, consider changing
-`dynamic` to `Object` instead.
-''',
-    );
-    _InheritedProviderScopeElement<T> inheritedElement;
+    _InheritedProviderScopeElement<T>? inheritedElement;
 
     if (context.widget is _InheritedProviderScope<T>) {
       // An InheritedProvider<T>'s update tries to obtain a parent provider of
       // the same type.
       context.visitAncestorElements((parent) {
         inheritedElement = parent.getElementForInheritedWidgetOfExactType<
-            _InheritedProviderScope<T>>() as _InheritedProviderScopeElement<T>;
+            _InheritedProviderScope<T>>() as _InheritedProviderScopeElement<T>?;
         return false;
       });
     } else {
       inheritedElement = context.getElementForInheritedWidgetOfExactType<
-          _InheritedProviderScope<T>>() as _InheritedProviderScopeElement<T>;
+          _InheritedProviderScope<T>>() as _InheritedProviderScopeElement<T>?;
     }
 
     if (inheritedElement == null) {
```

Comment:
Since the return type here is T? can you just return inheritedElement rather than throwing this exception? Otherwise the return type should logically be T as this will never return null due to the throw.